### PR TITLE
[Tags] Add a check for raw tag length

### LINKED ISSUE #568: [Tags] Better handle raw content exceeding max message length
## Summary

`[p]tag raw <tag name>` (as of https://github.com/SFUAnime/Ren/commit/646d6c9366ae8e1fded698cd2204b184f94d3183) fails when the raw content exceeds Discord's maximum message length. Note that the tag does exist because it does not exceed Discord's maximum message length, but the raw content of it (i.e., with markdown text formatters escaped) does.

## Reproduction

1. Create a tag where message length is as close to the message length limit as possible, and has many text formatters.
2. Get the raw content of that tag via `[p]tag raw <tag name>`.

## Potential solutions

Solutions are still being discussed, however two potential solutions are being considered:

1. The cog sends two split messages with no text wrapping (i.e., the first message's length is always at limit), or
2. The cog sends a text file containing the full raw content of the tag.

Open to ideas and feedback.

### PR DESCRIPTION
**Pull request by quachtridat**
## Summary

Add a check to prevent users from adding tags where the raw contents (markdown-escaped) would exceed Discord's max message length.

This fixes SFUAnime/Ren#568.

## Details

Add a helper function `checkLengthInRaw(str)` and constant `MAX_MSG_LEN` being 2000 (Discord's message length limit at the moment).

Check content length using that function before adding the content in:
- `[p]tag add/create`
- `[p]tag edit`
- `[p]tag generic`
- `[p]tag make`

### Unit testing

Unit tests for good/bad cases, with randomized strings of ASCII, non-ASCII and markdown-lookalike Japanese characters.

### Manual testing

Tested with a string of 1978 asterisks with `[p]tag edit`:
image

### CODE REVIEW

File: cogs/tags/testHelpers.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+                Utils.insertStringIntoString(
+                    "＞＞＞ ~*_`バーバラ＊いっくよ！～`_*~ ＜＜＜",
+                    Utils.randomNumberString(constants.MAX_MSG_LEN - 35),
+                    0,
+                ),
+            ),
+        ),
+    )
+    def testGoodCases(self, content: str):
+        assert helpers.checkLengthInRaw(content) == True
```
Comment (Injabie3) [inline cogs/tags/testHelpers.py]:
```suggestion
        assert helpers.checkLengthInRaw(content)
```

File: cogs/tags/testHelpers.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                Utils.insertStringIntoString(
+                    "＞＞＞ ~*_`バーバラ＊いっくよ！～`_*~ ＜＜＜",
+                    Utils.randomNumberString(constants.MAX_MSG_LEN - 34),
+                    0,
+                ),
+            ),
+        ),
+    )
+    def testBadCases(self, content: str):
+        assert helpers.checkLengthInRaw(content) == False
```
Comment (Injabie3) [inline cogs/tags/testHelpers.py]:
```suggestion
        assert not helpers.checkLengthInRaw(content)
```

**Review by Injabie3 — APPROVED** [review]
LOL those test cases. I was expecting maybe 2-4 to test basic working cases and near edge cases, but I guess this is fine.
